docs/guides/events/advanced.md

```diff
@@ -0,0 +1,115 @@
+# Advanced Usage
+
+```{attention}
+This guide only applies to GatewayBot. REST-only applications cannot receive events through the gateway.
+```
+
+## wait_for
+
+Sometimes you may want to wait for an event to be received in a procedural manner, then proceed
+with execution.
+
+With `wait_for()` you can block until you receive an event with a given predicate and timeout.
+
+In the example below, the bot prompts the user to play a number guessing game.
+Each iteration, the bot waits for the user to input a number, evaluates it, then gives an
+appropiate response.
+
+```py
+# We use random for number-generation and asyncio for exception handling
+import asyncio
+import random
+
+import hikari
+
+# -- Snip --
```

Comment:
Considering that this is a guide, and possibly one of the first things new users will read, I don't agree with leaving out bits and replacing them with `# -- Snip --`. At the very least, put this:
```suggestion
bot = hikari.GatewayBot(...)
```